source/server/http_test_server_filter.h

```diff
@@ -68,8 +68,11 @@ class HttpTestServerDecoderFilter : public Envoy::Http::StreamDecoderFilter {
                                     nighthawk::server::ResponseOptions& response_options);
 
 private:
+  void sendReply();
   const HttpTestServerDecoderFilterConfigSharedPtr config_;
   Envoy::Http::StreamDecoderFilterCallbacks* decoder_callbacks_;
+  nighthawk::server::ResponseOptions base_config_;
```

Comment:
Can you help me understand the lifetime of a HttpTestServerDecoderFilterConfig instance? Is it created just for one pair of request and reply?

I am trying to understand why we are storing the error message and the base config as instance attributes rather than local variables in a function.

Reply:
So I think `HttpTestServerDecoderFilterConfig` is what Envoy hands our filter instance, based on static (disk based) configuration, and possibly more sources. It's a shared pointer, so I assume these are shared across different instances of our filter. I guess that during configuration updates, two (or more) versions of `HttpTestServerDecoderFilterConfigPtr` are in existence, until all users of the old configuration(s) have dropped their references. Fortunately, this is probably transparant to our filter, and it seems we don't have to worry about configuration reloading as a result.

Our filter is instantiated per request, as far as I can tell, and therefore I added tracking of per-request state as instance attributes in there -- as we need to carry some information across various member calls.  

@htuch ^^ can you please verify this for correctness?

Reply:
Yeah, filter instance is per-request, config is shared.

Reply:
Thank you for the explanation.